lib/change-log.js

```diff
@@ -482,12 +476,16 @@ async function track_changes (req) {
     [ target, entityKey ] = await _prepareChangeLogForComposition(target, entityKey, changes, reqInfo)
   }
   const dbEntity = getDBEntity(target)
+
+
   await INSERT.into("sap.changelog.ChangeLog").entries({
     entity: dbEntity.name,
-    entityKey: entityKey,
+    entityKey: stringifyKey(entityKey),
     serviceEntity: target.name || target,
     changes: changes.filter(c => c.valueChangedFrom || c.valueChangedTo).map((c) => ({
       ...c,
+      serviceEntityPath: stringifyPath(c.serviceEntityPath),
```

Comment:
As discussed, if there is a single-key situation, we now serialize paths to the 'old' format before saving to the db.